materialize-postgres/sqlgen.go

```diff
@@ -22,20 +21,16 @@ func PostgresSQLGenerator() sql.Generator {
 			sql.STRING: sql.StringTypeMapping{
 				Default: sql.RawConstColumnType("TEXT"),
 				ByFormat: map[string]sql.TypeMapper{
-					// This format is for RFC3339 timestamps. As of this writing, Flow's schema
-					// validation does not actually validate that String values match their declared
-					// format, so there's no guarantee that the values will parse successfully here.
+					// According to the JSONSchema spec, this format is for RFC3339 timestamps, however
+					// since it seems a lot of usages of this format point to ISO8601 datetimes
+					// we have decided to support ISO8601 for this format as well.
+					// As of this writing, Flow's schema validation does not actually validate that
+					// String values match their declared format, so there's no guarantee that the values will parse successfully here.
 					"date-time": sql.ConstColumnType{
```

Comment:
This could be simplified to `"data-time": sql.RawConstColumnType("TIMESTAMPTZ")`, which always uses the identity function for the ValueConverter.

Reply:
@psFried I tried this on an actual materialization and it seemed to work. I actually realised to get the identity function you need to use `RawConstColumnType`, will do that